medicines/doc-index-updater/src/auth_manager/mod.rs

```diff
@@ -0,0 +1,136 @@
+extern crate base64;
+
+use futures::future;
+use regex::Regex;
+use warp::{Filter, Rejection};
+
+#[derive(Clone, Debug)]
+pub struct AuthenticationFailed;
+impl warp::reject::Reject for AuthenticationFailed {}
+
+fn get_basic_username() -> String {
+    std::env::var("BASIC_AUTH_USERNAME").expect("Set env variable BASIC_AUTH_USERNAME first!")
+}
+
+fn get_basic_password() -> String {
+    std::env::var("BASIC_AUTH_PASSWORD").expect("Set env variable BASIC_AUTH_PASSWORD first!")
+}
+
+fn auth_is_correct(username: String, password: String) -> bool {
+    username == get_basic_username() && password == get_basic_password()
+}
+
+fn extract_encoded_credentials(auth_header: String) -> Option<String> {
+    let re = Regex::new(r"^Basic\s(?P<encoded_credentials>[-A-Za-z0-9+/]*={0,3})$").unwrap();
+
+    if let Some(caps) = re.captures(&auth_header) {
+        match caps.name("encoded_credentials") {
+            Some(creds) => Some(creds.as_str().to_string()),
+            _ => None,
+        }
+    } else {
+        None
+    }
+}
+
+fn decode_credentials(encoded_credentials: String) -> Option<(String, String)> {
+    if let Ok(credentials) = base64::decode(encoded_credentials) {
+        let re = Regex::new(r"^(?P<username>\w+):(?P<password>\w+)$").unwrap();
```

Comment:
this Regex is going to get interpreted every call, right? can we make it created just once? might be a premature optimisation, though

Reply:
If it becomes a bottleneck it's possible. Would prefer to avoid global / magic state though unless it's unavoidable.